src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -248,6 +253,30 @@ public static ReflectionAIFunctionDescriptor GetOrCreate(MethodInfo method, AIFu
 
         private ReflectionAIFunctionDescriptor(DescriptorKey key, JsonSerializerOptions serializerOptions)
         {
+            AIJsonSchemaCreateOptions schemaOptions = new()
+            {
+                // This needs to be kept in sync with the shape of AIJsonSchemaCreateOptions.
```

Comment:
Should AIJsonSchemaCreateOptions expose a Clone method? Or be a record if that won't introduce other issues?

Reply:
Given that `AIJsonSchemaCreateOptions` has custom implementations for `Equals` and `GetHashCode` that seem to mirror what records would give us, I think that makes sense. Especially now if we're cloning in some places.

If we don't go the record route, it would probably be best to extract this clone into an internal testable method, so you could write a reflection-based test like [this](https://github.com/dotnet/aspnetcore/blob/18b3b4ac8dfc4498edc6bf26c7bde9a35ef840f8/src/Servers/Kestrel/Core/test/SniOptionsSelectorTests.cs#L807) to verify no one accidentally adds a property and forgets to update the clone method.